habitLoggerApp.Natephor/HabitManager.cs

```diff
@@ -0,0 +1,147 @@
+    {
+        using var connection = new SQLiteConnection(ConnectionString);
+        connection.Open();
+
+        var insertQuery =
+            """
+            INSERT INTO habit (Date, Body, Quantity) 
+                VALUES (@date, @body, @quantity)
+            """;
+        var command = connection.CreateCommand();
+        command.CommandText = insertQuery;
+        command.Parameters.Add(new SQLiteParameter("@date", date));
+        command.Parameters.Add(new SQLiteParameter("@body", body));
+        command.Parameters.Add(new SQLiteParameter("@quantity", quantity));
+
+        try
+        {
+            command.ExecuteNonQuery();
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine(ex.Message);
+        }
+    }
+
+    public bool UpdateHabit(int id, string body, string date, int quantity)
+    {
+        using var connection = new SQLiteConnection(ConnectionString);
+        connection.Open();
+
+        var updateQuery =
+            """
+            UPDATE habit
+            SET Body = @body, Date = @date, Quantity = @quantity
+            WHERE Id = @id
+            """;
+
+        var command = connection.CreateCommand();
+        command.CommandText = updateQuery;
+        command.Parameters.Add(new SQLiteParameter("@id", id));
+        command.Parameters.Add(new SQLiteParameter("@body", body));
+        command.Parameters.Add(new SQLiteParameter("@date", date));
+        command.Parameters.Add(new SQLiteParameter("@quantity", quantity));
+
+        var updatedCount = 0;
+        try
+        {
+            updatedCount = command.ExecuteNonQuery();
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine(ex.Message);
+        }
+
+        return updatedCount > 0;
+    }
+
+    public bool DeleteHabit(int id)
+    {
+        using var connection = new SQLiteConnection(ConnectionString);
+        connection.Open();
+
+        var deleteQuery =
+            """
+            DELETE FROM habit
+            WHERE Id = @Id;
+            """;
+
+        var command = connection.CreateCommand();
+        command.CommandText = deleteQuery;
+        command.Parameters.Add(new SQLiteParameter("@Id", id));
+
+        var rowCount = 0;
+        try
+        {
+            rowCount = command.ExecuteNonQuery();
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine(ex.Message);
+        }
+
+        return rowCount > 0;
+    }
+
+    public List<Habit> GetAllHabits()
+    {
+        var habits = new List<Habit>();
+        using var connection = new SQLiteConnection(ConnectionString);
+        connection.Open();
+
+        var command = connection.CreateCommand();
+        command.CommandText =
+            """
+            SELECT Id, Date, Body, Quantity FROM habit
+            """;
+
+        try
+        {
+            using var reader = command.ExecuteReader();
+            if (reader.HasRows)
```

Comment:
🟡 **Readability**

💡 Contensious one here, so marked in yellow. But, `reader.Read()` will only be true if `reader.HasRows`, so one could argue that the `reader.HasRows` only introduces a redundant layer of indentation.